src/daemon/LogindDBusTypes.cpp

```diff
@@ -46,17 +48,26 @@ LogindPathInternal::LogindPathInternal()
     qRegisterMetaType<UserInfoList>("UserInfoList");
     qDBusRegisterMetaType<UserInfoList>();
 
-    if (QDBusConnection::systemBus().interface()->isServiceRegistered(QStringLiteral("org.freedesktop.login1"))) {
-        qDebug() << "Logind interface found";
-        available = true;
-        serviceName = QStringLiteral("org.freedesktop.login1");
-        managerPath = QStringLiteral("/org/freedesktop/login1");
-        managerIfaceName = QStringLiteral("org.freedesktop.login1.Manager");
-        seatIfaceName = QStringLiteral("org.freedesktop.login1.Seat");
-        sessionIfaceName = QStringLiteral("org.freedesktop.login1.Session");
-        userIfaceName = QStringLiteral("org.freedesktop.login1.User");
-        return;
+#ifdef HAVE_SYSTEMD
+    // systemd-logind should be running, although because it takes a few moments to restart after
+    // systemctl isolate calls, it may not yet be running. Wait a few seconds for it, while blocking everything else.
+    int logind_wait_seconds = 50;
+    while (logind_wait_seconds--) {
```

Comment:
Could we just change the isServiceRegistered to startService?
```
auto reply = QDBusConnection::systemBus().interface()->startService(QStringLiteral("org.freedesktop.login1"));
if (reply.isValid()) {
```
This way we would be able to remove that ifdef and loop, they are a bit odd...

Reply:
I've put the proposal here for you to test: https://github.com/aleixpol/sddm/tree/delay-seat-init

Reply:
Hey, so the problem isn't that logind isn't starting, its that while sddm is marked as dependent in logind, it looked like systemd was getting ahead of itself and sddm was starting before logind was registered on dbus. Hence the delay, is enough to slow sddm down enough for logind to arrive.

I opened a github issue about it: but have not returned to the problem. In part because the simpledrm driver has largly fixed the core issue and i'm a bit confused about how it was happening too, despite having clear evidence of such, so maybe its was just some kind of dbus ordering issue.

https://github.com/systemd/systemd/issues/22022

Reply:
Okay, then let's see how this resolves in systemd and we can follow up on this later on.